x-pack/legacy/plugins/apm/public/components/app/ServiceOverview/ServiceList/index.tsx

```diff
@@ -50,7 +51,10 @@ export const SERVICE_COLUMNS = [
     sortable: true,
     render: (serviceName: string) => (
       <EuiToolTip content={formatString(serviceName)} id="service-name-tooltip">
-        <AppLink path={`/${serviceName}/transactions`}>
+        <AppLink
+          path={`/services/${serviceName}/transactions`}
+          documentType={DOCUMENT_TYPE.SERVICES}
```

Comment:
We cannot assign a single "document type" to a link. Eg. We have two types (transactions and spans) on the transaction details page. 
Similarly the error list page and error details page both show errors but could have different filters. 

The filters we show are dependent on the route - so if anything we could keep the filter info in the route config. However, I think this ties back to the talk we had about improving linking and making them type safe is the long term solution.

I therefore think we should use explicit query params in links, and get the type safety added as quickly as possible.

Reply:
Ah, maybe then `documentType` should be something like `viewType`? When passing explicit query params, do you mean hard-coding them as props to AppLink? What would then happen if we add or remove a local filter from the configuration? In this case the query prop is pretty generic so hard to have TypeScript help us out then. 

I do think we can add the configuration to the route though.. and then we could use a utility function to pass the right query parameters to `APMLink`.

Reply:
> When passing explicit query params, do you mean hard-coding them as props to AppLink?

I'm thinking something like this:

```js
        <AppLink
          routeName="transactionList"
          params={{ serviceName, hostId }}
        >  
```

If we can infer the required and optional `params` from the `routeName`

> What would then happen if we add or remove a local filter from the configuration

I think the route config should decide what you can pass into links - not the filter config. Those two should be decoupled imo.